sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1911,452 @@ def transform(self, X):
+            The data used to scale along the features axis. If a sparse
+            matrix is provided, it will be converted into a sparse
+            ``csc_matrix``. Additionally, the sparse matrix needs to be
+            semi-positive.
+
+        Returns
+        -------
+        self : object
+            Returns self
+        """
+        X = check_array(X, accept_sparse='csc')
+
+        if self.n_quantiles <= 0:
+            raise ValueError("Invalid value for 'n_quantiles': %d. "
+                             "The number of quantiles must be at least one."
+                             % self.n_quantiles)
+
+        if self.subsample <= 0:
+            raise ValueError("Invalid value for 'subsample': %d. "
+                             "The number of quantiles must be at least one."
+                             % self.subsample)
+
+        # we only accept positive sparse matrix
+        if sparse.issparse(X) and np.any(X.data < 0):
+            raise ValueError('QuantileTransformer only accepts non-negative'
+                             ' sparse matrices')
+
+        if sparse.issparse(X):
+            self._sparse_fit(X)
+        else:
+            self._dense_fit(X)
+
+        self._build_f()
+
+        return self
+
+    def _dense_transform(self, X, direction=True):
+        """Forward and inverse transform for dense matrices.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            The data used to scale along the features axis.
+
+        direction : bool, optional (default=True)
+            If True, apply forward transform. If False, apply
+            inverse transform.
+
+        Returns
+        -------
+        X : ndarray, shape (n_samples, n_features)
+            Projected data.
+        """
+        if direction:
+            func_transform = self._f_transform
+        else:
+            func_transform = self._f_inverse_transform
+        output_distribution = getattr(stats, self.output_distribution)
+
+        references = np.linspace(0, 1, self.n_quantiles, endpoint=True)
+
+        for feature_idx, f in enumerate(func_transform):
+            # older version of scipy do not handle tuple as fill_value
+            # clipping the value before transform solve the issue
+            if direction:
+                lower_bound_x = self.quantiles_[0, feature_idx]
+                upper_bound_x = self.quantiles_[-1, feature_idx]
+                lower_bound_y = references[0]
+                upper_bound_y = references[-1]
+            else:
+                lower_bound_x = references[0]
+                upper_bound_x = references[-1]
+                lower_bound_y = self.quantiles_[0, feature_idx]
+                upper_bound_y = self.quantiles_[-1, feature_idx]
+            if not direction:
+                #  for inverse transform, match a uniform PDF
+                for i in range(X.shape[0]):
+                    X[i, feature_idx] = output_distribution.cdf(
```

Comment:
https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.rv_continuous.cdf.html#scipy.stats.rv_continuous.cdf suggests it does (and even version 0.9.0).